pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statuspeer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	netv1alpha1 "github.com/liqotech/liqo/apis/net/v1alpha1"
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/liqoctl/output"
+	"github.com/liqotech/liqo/pkg/liqoctl/status"
+	liqonetutils "github.com/liqotech/liqo/pkg/liqonet/utils"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+	"github.com/liqotech/liqo/pkg/utils/getters"
+	peeringconditionsutils "github.com/liqotech/liqo/pkg/utils/peeringConditions"
+	"github.com/liqotech/liqo/pkg/utils/share"
+	"github.com/liqotech/liqo/pkg/utils/slice"
+)
+
+// PeerInfoChecker implements the Check interface.
+// holds the information about the peered cluster.
+type PeerInfoChecker struct {
+	options          *Options
+	peerInfoSection  output.Section
+	collectionErrors []status.CollectionError
+}
+
+const (
+	peerInfoCheckerName = "Peered Cluster Information"
+)
+
+// newPeerInfoChecker return a new PeerInfoChecker.
+func newPeerInfoChecker(options *Options) *PeerInfoChecker {
+	return &PeerInfoChecker{
+		peerInfoSection: output.NewRootSection(),
+		options:         options,
+	}
+}
+
+// Silent implements the Check interface.
+func (pic *PeerInfoChecker) Silent() bool {
+	return false
+}
+
+// Collect implements the collect method of the Checker interface.
+// it collects the infos of the peered cluster.
+func (pic *PeerInfoChecker) Collect(ctx context.Context) error {
+	localClusterIdentity, err := liqoutils.GetClusterIdentityWithControllerClient(ctx, pic.options.CRClient, pic.options.LiqoNamespace)
+	localClusterName := localClusterIdentity.ClusterName
+	if err != nil {
+		pic.addCollectionError("LocalClusterName", "", err)
+	}
+
+	tunnelEndpoints, err := getters.GetTunnelEndpointsByLabel(ctx, pic.options.CRClient, labels.NewSelector())
```

Comment:
I'm wondering why you iterate over the tunnelendpoints instead of the foreign clusters, which would be much more natural.